### ISSUE TITLE
Restructure dunner file to incorporate global variables/arguments 

### ISSUE THREAD

**Issue by ayushjn20** [leopardslab/dunner#146]
**Is your feature request related to a problem? Please describe.**
Sometimes it is required for the user to use the same environment variables over multiple steps and/or multiple tasks.      

**Describe the solution you'd like**
There should be a way of defining different levels of scope of the variables. The variable can be defined for all the tasks, or all the steps of task, and so on. The variable value can be overridden based on the scope.
Also, it is required that some default arguments must be passed to all the steps of the task, through which, if user does not define the required nth argument, `$n` takes `defaultArgs[n]` value.

**Comment by agentmilindu** [comment]
A sample `.dunner.yaml` would be,

```
envs:
    - AWS_ACCESS_KEY_ID=`$AWS_KEY`
    - AWS_SECRET_ACCESS_KEY=`$AWS_SECRET`
    - AWS_DEFAULT_REGION=us-east1
mount:
    - '~/.aws:~.aws:r'
tasks:
    status:
        steps:
            - image: mesosphere/aws-cli
              commands: 
                - [aws, 'elasticbeanstalk describe-events --environment-name $1']
    deploy:
        envs:
            - API_BASE=api.example.com
        steps:
            - image: emeraldsquad/sonar-scanner, 
              commands: 
                - [sonar, scan]
            - image: golang
              commands: 
                - [go, install]
            - image: mesosphere/aws-cli
              commands: 
                - [aws, 'elasticbeanstalk update-application --application-name myapp']
```

**Comment by ayushjn20** [comment]
Closed with reference to #148
